File: Word Counter/Python/counter.py
import time

def count_e_in_file(filename):
    try:
        with open(filename, 'r') as file:
            text = file.read()
            text = text.lower()
            count = text.count('e')
            return count
    except FileNotFoundError:
        return 0

file_path = '../bee-movie.txt'

start_time = time.time()
count = count_e_in_file(file_path)
end_time = time.time()
execution_time = end_time - start_time
print(f"Number of 'e' characters (non-case-sensitive) in the file: {count}")
print(f"Execution time: {execution_time} seconds")

File: Word Counter/Go/counter.go
package main

import (
	"fmt"
	"os"
	"strings"
	"time"
)

func countEInFile(filename string) int {
	content, err := os.ReadFile(filename)
	if err != nil {
		return 0
	}

	text := string(content)
	text = strings.ToLower(text)
	count := strings.Count(text, "e")
	return count
}

func main() {
	filePath := "../bee-movie.txt"

	startTime := time.Now()
	count := countEInFile(filePath)
	endTime := time.Now()
	executionTime := endTime.Sub(startTime).Seconds()

	fmt.Printf("Number of 'e' characters (non-case-sensitive) in the file: %d\n", count)
	fmt.Printf("Execution time: %f seconds\n", executionTime)
}
